packages/back-end/src/models/DataSourceModel.ts

```diff
@@ -336,6 +343,71 @@ export async function updateDataSource(
     return;
   }
 
+  if (
+    isDataSourceType<GrowthbookClickhouseDataSource>(
+      datasource,
+      "growthbook_clickhouse"
+    ) &&
+    isPartialWithMaterializedColumns(updates)
+  ) {
+    updates.settings!.materializedColumns = sanitizeMaterializedColumns(
```

Comment:
Instead of `updates.settings!` everywhere, can we just put that into the `if` condition?  `if (... && updates.settings && updates.settings.materializedColumns)`.

Reply:
Actually, looking at this more, I think we should make dedicated CRUD endpoints and methods instead of using the generic update route for everything.  
- Code will be much cleaner and easier to follow
- Avoids race conditions (2 people changing different parts of the materializedColumns array at the same time)
- Avoids unintentional changes (we had something similar recently in environments caused by passing the whole array to the back-end. A bug meant re-ordering the array would sometimes delete things by accident)
- It will map much better to the front-end actions (click the "delete" button, it calls a dedicated "delete" endpoint)